src/main/java/com/robinpowered/sdk/RobinApi.java

```diff
@@ -134,6 +135,16 @@ public boolean hasCredential() {
     }
 
     /**
+     * Returns whether a the interface is authenticated with a valid credential.
+     * Valid credentials have non-null values.
+     *
+     * @return Whether a valid credential has been set.
+     */
+    public boolean isAuthenticated() {
+        return credential != null && credential.getValue() != null;
```

Comment:
That could work, but to me personally it seems unintuitive. 
Seems like we just need better naming for my method. Would `hasAuthentication` be better?